### PR TITLE
feat: add target constrain to POST and PATCH apis

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

add target constrain to POST and PATCH apis.

## Testing

Passed UTs and ITs.

### CODE REVIEW

File: packages/phrases/src/locales/zh-cn/errors.ts [hunk 1]
```diff
@@ -102,6 +102,9 @@ const errors = {
     invalid_type_for_syncing_profile: '只有社交连接器可以开启用户档案同步。',
     can_not_modify_target: '不可修改连接器 target。',
     multiple_target_with_same_platform: '不能同时存在多个有相同 target 和平台类型的社交连接器。',
+    empty_target_for_standard_social_connector: '对于标准社交连接器，缺少 target 参数。',
+    cannot_change_target_for_non_standard_social_connector:
```
Comment (wangsijie) [inline packages/phrases/src/locales/zh-cn/errors.ts:106]: 不可修改

File: packages/core/src/routes/connector.ts [hunk 2]
```diff
@@ -129,6 +129,27 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
         });
       }
 
+      const isStandardSocial =
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:132]: is `isStandard` enough for us?

File: packages/core/src/routes/connector.ts [hunk 3]
```diff
@@ -129,9 +129,26 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
         });
       }
 
+      const isStandard = connectorFactory.metadata.isStandard === true;
+      const hasConfigurableMetadata = configurableMetadata !== undefined;
+      const cleanedDatabaseMetadata = cleanDeep(configurableMetadata);
+
+      if (isStandard) {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:136]: try to use 2 `assetThat` to replace this `if` block?

File: packages/core/src/routes/connector.ts [hunk 4]
```diff
@@ -140,25 +157,23 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
       if (connectorFactory.type === ConnectorType.Social) {
         const connectors = await getLogtoConnectors();
-        const connectorTarget = body.metadata?.target ?? connectorFactory.metadata.target;
         assertThat(
           !connectors
             .filter(({ type }) => type === ConnectorType.Social)
             .some(
               ({ metadata: { target, platform } }) =>
-                target === connectorTarget && platform === connectorFactory.metadata.platform
+                target === cleanedDatabaseMetadata?.target &&
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:165]: the original target "fallback" logic is more powerful?

File: packages/core/src/routes/connector.ts [hunk 5]
```diff
@@ -197,14 +212,29 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
         body,
       } = ctx.guard;
+      const hasConfigurableMetadata = configurableMetadata !== undefined;
+      const cleanedDatabaseMetadata = cleanDeep(configurableMetadata);
 
       const { type, validateConfig, metadata } = await getLogtoConnectorById(id);
 
       const isStandard = metadata.isStandard === true;
 
+      if (isStandard) {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:225]: we can simply prevent target modify for all type of connectors?

File: packages/phrases/src/locales/en/errors.ts [hunk 6]
```diff
@@ -111,6 +111,8 @@ const errors = {
     can_not_modify_target: 'The connector target can not be modified.',
     multiple_target_with_same_platform:
       'You can not have multiple social connectors that have same target and platform.',
+    cannot_change_metadata_for_non_standard_connector:
+      "Should not config 'metadata' for a non standard connector.",
```
Comment (charIeszhao) [inline packages/phrases/src/locales/en/errors.ts:115]: IMO, we don't necessarily need to expose the `standard connector` concept to any end users. What about just `Connector metadata cannot be changed.`

File: packages/phrases/src/locales/de/errors.ts [hunk 7]
```diff
@@ -112,6 +112,8 @@ const errors = {
     can_not_modify_target: 'The connector target can not be modified.',
     multiple_target_with_same_platform:
       'You can not have multiple social connectors that have same target and platform.',
+    cannot_change_metadata_for_non_standard_connector:
+      "Should not config 'metadata' for a non standard connector.",
```
Comment (charIeszhao) [inline packages/phrases/src/locales/de/errors.ts:116]: nit: Need to add `// UNTRANSLATED` annotation to these phrases?

File: packages/core/src/routes/connector.ts [hunk 8]
```diff
@@ -129,9 +129,26 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
         });
       }
 
+      const isStandard = connectorFactory.metadata.isStandard === true;
+      const hasConfigurableMetadata = configurableMetadata !== undefined;
```
Comment (charIeszhao) [inline packages/core/src/routes/connector.ts:133]: These two are not necessary

File: packages/core/src/routes/connector.ts [hunk 9]
```diff
@@ -129,9 +129,26 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+          'connector.can_not_modify_target'
+        );
+      } else {
+        assertThat(
+          !hasConfigurableMetadata,
+          'connector.cannot_change_metadata_for_non_standard_connector'
+        );
+      }
+
       const { count } = await countConnectorByConnectorId(connectorId);
```
Comment (charIeszhao) [inline packages/core/src/routes/connector.ts:149]: nit: `getConnectorCountByConnectorId`

File: packages/core/src/routes/connector.ts [hunk 10]
```diff
@@ -197,11 +203,21 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
       const {
         params: { id },
-        body: { config, syncProfile, metadata },
+        body: { config, metadata, syncProfile },
       } = ctx.guard;
 
       const { type, validateConfig, metadata: originalMetadata } = await getLogtoConnectorById(id);
 
+      assertThat(
+        originalMetadata.isStandard !== true || metadata?.target === originalMetadata.target,
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:212]: ditto

**Pull request merged**